deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java

```diff
@@ -101,15 +108,79 @@ QuarkusHillaEnvironmentBuildItem detectQuarkusHillaMode(CurateOutcomeBuildItem o
+            BuildProducer<AdditionalIndexedClassesBuildItem> additionalClasses) {
+        if (capabilities.isPresent(SPRING_DATA_SUPPORT)) {
             producer.produce(new ExcludeDependencyBuildItem("com.github.mcollovati", "hilla-shaded-deps"));
+            additionalClasses.produce(new AdditionalIndexedClassesBuildItem(
+                    com.github.mcollovati.quarkus.hilla.crud.spring.FilterableRepository.class.getName(),
+                    FilterableRepositorySupport.class.getName()));
+        }
+        if (capabilities.isPresent(PANACHE_SUPPORT)) {
+            additionalClasses.produce(new AdditionalIndexedClassesBuildItem(
+                    com.github.mcollovati.quarkus.hilla.crud.panache.FilterableRepository.class.getName(),
+                    FilterableRepositorySupport.class.getName()));
+        }
+    }
+
+    @BuildStep
+    void implementSpringDataFilterableRepositoryExtension(
+            Capabilities capabilities,
+            CombinedIndexBuildItem index,
+            BuildProducer<BytecodeTransformerBuildItem> byteCodeProducer) {
+        if (capabilities.isPresent(SPRING_DATA_SUPPORT)) {
+            IndexView indexView = index.getComputingIndex();
+            DotName filterableRepositoryInterface = DotName.createSimple(FilterableRepository.class);
+            FilterableRepositoryImplementor visitorFunction =
+                    new FilterableRepositoryImplementor(indexView, filterableRepositoryInterface);
+            indexView
+                    .getKnownDirectImplementors(filterableRepositoryInterface)
+                    .forEach(ci -> byteCodeProducer.produce(new BytecodeTransformerBuildItem.Builder()
+                            .setClassToTransform(ci.name().toString())
+                            .setVisitorFunction(visitorFunction)
+                            .build()));
         }
     }
+
+    @BuildStep
+    void implementPanacheFilterableRepositoryExtension(
+            Capabilities capabilities,
+            CombinedIndexBuildItem index,
+            BuildProducer<BytecodeTransformerBuildItem> producer) {
+        if (capabilities.isPresent(PANACHE_SUPPORT)) {
+            IndexView indexView = index.getComputingIndex();
+            DotName filterableRepositoryInterface =
+                    DotName.createSimple(com.github.mcollovati.quarkus.hilla.crud.panache.FilterableRepository.class);
+            FilterableRepositoryImplementor visitorFunction =
+                    new FilterableRepositoryImplementor(indexView, filterableRepositoryInterface);
+            indexView
+                    .getKnownDirectImplementors(filterableRepositoryInterface)
+                    .forEach(ci -> producer.produce(new BytecodeTransformerBuildItem.Builder()
+                            .setClassToTransform(ci.name().toString())
+                            .setVisitorFunction(visitorFunction)
+                            .build()));
+        }
+    }
+
```

Comment:
Factor out additional method to avoid duplication?

Reply:
Sounds like a good idea

Reply:
Refactored by introducing a BuildItem to collect classes to augment and a single method that enqueues the byte code production.